hivemind/optim/collaborative.py

```diff
@@ -354,7 +354,7 @@ def report_training_progress(self):
             with self.lock_local_progress:
                 current_time = get_dht_time()
                 local_state_info = TrainingState(
-                    endpoint=self.averager.endpoint,
+                    peer_id=self.averager.endpoint.to_base58(),
```

Comment:
I'm slightly against casting to base58 in multiple places all over the code; would appreciate if it was possible to come up with a way to reduce this casting :)

Or maybe you can just call `__str__` everywhere, since `to_base58` is an implementation detail

Reply:
We can use `str(endpoint)`, however we would still need to call `PeerID.from_base58(value)` to deserialize. Therefore, I would keep these operations symmetric.

Also, it is actually more natural to use `bytes` for representing `PeerID`s in protobufs (and change `endpoint.to_base58()`/`PeerID.from_base58(value)` to `endpoint.to_bytes()`/`PeerID(value)`). However, DHT already uses `str` for `PeerIDs` in protobufs, so I'd like to make the code consistent in this PR (but I don't mind changing everything to `bytes` in a separate PR).

Reply:
Sure, let's change it in a follow-up